internal/provisioner/objects/object.go

```diff
@@ -47,16 +47,16 @@ func NewUnprivilegedPodSecurity() *corev1.PodSecurityContext {
 	}
 }
 
-// ObjectGetter gets an object given a namespace and name.
-type ObjectGetter func(ctx context.Context, cli client.Client, namespace, name string) (client.Object, error)
-
-// ObjectUpdater update the current resource to desired if need.
-type ObjectUpdater func(ctx context.Context, cli client.Client, contour *model.Contour, current, desired client.Object) error
-
 // EnsureObjectDeleted ensures that an object with the given namespace and name is deleted
 // if Contour owner labels exist.
-func EnsureObjectDeleted(ctx context.Context, cli client.Client, contour *model.Contour, name string, getter ObjectGetter) error {
-	obj, err := getter(ctx, cli, contour.Namespace, name)
+func EnsureObjectDeleted[T client.Object](
+	ctx context.Context,
+	cli client.Client,
+	contour *model.Contour,
+	name string,
+	getObject func(ctx context.Context, cli client.Client, namespace, name string) (T, error),
```

Comment:
we can probably get rid of passing the `getObject` func in here since all the implementations are the same just issuing a `client.Get(...)` call, will mess around some more.